WcaOnRails/app/controllers/results_submission_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'fileutils'
+
+class ResultsSubmissionController < ApplicationController
+  before_action :authenticate_user!
+  before_action -> { redirect_to_root_unless_user(:can_submit_competition_results?, competition_from_params) }
+
+  def edit
+    @competition = competition_from_params
+  end
+
+  def submit
+    @competition = competition_from_params
+
+    if params[:results].blank? || params[:message].blank?
+      flash.now[:danger] = "Please make sure to fill in the message and attach the results file."
+      render :edit
```

Comment:
The text is still there, expect the line breaks, interestingly.
It renders as "Line1Line2Line3".
I can try to fix that when I'm done with the more important stuff.

Reply:
I dug into this, it turns out it's because you're using `text_field_tag` instead of `text_area_tag`. I guess text fields cannot have newlines of any sort. It would be nice if simplemde complained about being created on a text field! Could you file an issue on https://github.com/sparksuite/simplemde-markdown-editor/issues asking about having simplemde complain if it's being run against a non textarea?